### PR TITLE
Early work on supporting some static analysis checks in the APIView tool for Java

### PR DESCRIPTION
**Pull request by JonathanGiles**
Currently, the checks I have created are the following:

**Package names:**
 * Must start with 'com.azure'.
 * no uppercase, no underscores, no hyphens.

**Illegal imports:**
 * no imports of 'com.sun' packages

**All public API:**
 * No external dependencies exposed through method return type or parameters
 * No implementation APIs exposed through method return type or parameters
 * No non-static public or protected fields

### CODE REVIEW

**Comment by JonathanGiles** [comment]
@pakrym This upgrades the version to 1.1.0, which means that the .net code that calls the compiled library also needs to be updated to reference the new version in the file name.

File: src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/PackageNameDiagnosticRule.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import static com.azure.tools.apiview.processor.analysers.util.ASTUtils.*;
+
+public class PackageNameDiagnosticRule implements DiagnosticRule {
+
+    @Override
+    public void scan(final CompilationUnit cu, final APIListing listing) {
+        getPackageName(cu).ifPresent(packageName -> {
+            // we need to map the issue to the class id, because package text isn't printed in the APIView output
+            getClassName(cu).map(listing.getKnownTypes()::get).ifPresent(typeId -> {
+                if (!packageName.startsWith("com.azure")) {
```
Comment (srnagar) [inline src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/PackageNameDiagnosticRule.java]: You can use the regex below instead to ensure package names are valid. So, we will not need `if` conditions for each of the special chars.

`^com.azure.[a-z0-9]*(\.[a-z0-9]+)+[0-9a-z]$`

Tested here - https://regex101.com/r/eE4aY9/33

Comment (JonathanGiles) [reply src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/PackageNameDiagnosticRule.java]: Thanks!

**Comment by JonathanGiles** [comment]
@pakrym This is now merged. You can do a release whenever you want.

**Pull request merged**